lncrawl/sources/novelfull.py

```diff
@@ -122,14 +123,33 @@ def download_chapter_body(self, chapter):
         '''Download body of a single chapter and return as clean html format.'''
         logger.info('Downloading %s', chapter['url'])
         soup = self.get_soup(chapter['url'])
+
         content = soup.select_one('div#chapter-content')
+
+        # Removes junk text from chapters.
+        self.blacklist_patterns = [
+            r'^Translator:',
+            r'^Editor:',
+            r'^Atlas Studios',
+            r'Read more chapter on NovelFull'
+            r'^full thich ung',
+        ]
+
+        # Some comments in source code of site gets converted into text/paragraphs. This removes it.
+        for comment in soup.findAll(text=lambda text:isinstance(text, Comment)):
```

Comment:
Since you're adding a custom function here, could you move the below `findAll`s and `select`s here?
```py
def is_ad(tag):
    return isinstance(tag, Comment) or (
        tag.name == "div" and tag.get("class") in [["left"], ["center"]]
    )
for ads in content.find_all(is_ad):
    ads.decompose()
```
Also please use `find_all` https://www.python.org/dev/peps/pep-0008/